frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx

```diff
@@ -16,39 +20,58 @@ const ProjectAccessForm: React.FC<ProjectAccessFormProps> = ({
   errors,
   dirty,
   roles,
+  roleBindings,
+  values,
 }) => {
   const { t } = useTranslation();
+  const [isStaleInfo, setIsStaleInfo] = React.useState<boolean>(false);
   const disableSubmit = !dirty || !_.isEmpty(errors);
+
+  React.useEffect(() => {
+    if (!_.isEqual(roleBindings, values)) {
+      setIsStaleInfo(true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [roleBindings]);
+
+  const onReload = () => {
+    handleReset();
+    setIsStaleInfo(false);
+  };
+
   return (
     <Form onSubmit={handleSubmit}>
-      <div className="co-m-pane__form">
-        <MultiColumnField
-          name="projectAccess"
-          addLabel={t('devconsole~Add access')}
-          headers={[t('devconsole~Name'), t('devconsole~Role')]}
-          emptyValues={{ user: '', role: '' }}
-        >
-          <InputField name="user" type={TextInputTypes.text} placeholder={t('devconsole~Name')} />
-          <DropdownField
-            name="role"
-            title={t('devconsole~Select a role')}
-            items={roles}
-            fullWidth
-          />
-        </MultiColumnField>
+      <div className="co-m-pane__body">
+        <div className="co-m-pane__form">
+          <MultiColumnField
+            name="projectAccess"
+            addLabel={t('devconsole~Add access')}
+            headers={[t('devconsole~Name'), t('devconsole~Role')]}
+            emptyValues={{ user: '', role: '' }}
+          >
+            <InputField name="user" type={TextInputTypes.text} placeholder={t('devconsole~Name')} />
+            <DropdownField
+              name="role"
+              title={t('devconsole~Select a role')}
+              items={roles}
+              fullWidth
+            />
+          </MultiColumnField>
+        </div>
 
         <hr />
-        <FormFooter
-          handleReset={handleReset}
-          isSubmitting={isSubmitting}
-          errorMessage={status && status.submitError}
-          successMessage={status && !dirty && status.success}
-          disableSubmit={disableSubmit}
-          showAlert={!disableSubmit}
-          submitLabel={t('devconsole~Save')}
-          resetLabel={t('devconsole~Reload')}
-        />
       </div>
+      <FormFooter
+        handleReset={onReload}
+        isSubmitting={isSubmitting}
+        errorMessage={status && status.submitError}
+        successMessage={status && !dirty && status.success}
+        disableSubmit={disableSubmit}
+        showAlert={!disableSubmit}
+        submitLabel={t('devconsole~Save')}
+        resetLabel={t('devconsole~Reload')}
+        showStaleAlert={isStaleInfo}
```

Comment:
I dont think you need to add a new alert element in FormFooter for displaying this alert as it is an information alert which we already have in FormFooter. If the rolebindings have updated in the background we need to show only the  stale-info alert.
```suggestion
        showAlert={isStaleInfo || !disableSubmit}
        infoTitle={isStaleInfo && <title>}
        infoMessage={isStaleInfo && <message>}
```

Also if you want to have a separate alert for this, though there is no need at the moment, you can add a new prop to FormFooter that accepts an array of custom alert(s) and render it below the generic info-alert rather than adding a specific prop for each alert as there might be other custom alerts we want to show later in some use-case. FormFooter shouldn't be bothered to handle the custom alerts.

Reply:
I tried to follow what we do for the yaml editor. I wanted to show both the info alerts (`You made changes to this page.` & `This object has been updated.`) which is why I went ahead with a separate prop for showing a stale alert. But I could try out what you suggested i.e creating an array of custom alerts

Few questions here:
- do you expect the `Save` button to be disabled if the displayed list is stale?
- right now when we make changes in the form it shows an info alert saying changes have been made to the form. So, when the stale info alert is shown should we hide the above alert?

cc @serenamarie125 @christianvogt @jerolimov

Reply:
Yes, I think it makes sense to disable save when the currently shown data becomes stale as the user should not update it because it might cause conflicts. Also if we show stale-info alert, as I have mentioned above also, the user should not update the stale data so we shouldn't show obj-updated alert along stale-info alert. Since we shouldn't have both alerts showing at the same time I think we go can go with the first approach I suggested for this usecase.